seed/challenges/08-coding-interview-questions-and-take-home-assignments/coding-interview-algorithm-questions.json

```diff
@@ -448,6 +447,185 @@
           ]
         }
       }
+    },
+    {
+      "id": "8d5123c8c441eddfaeb5bdef",
+      "title": "Implement Bubble Sort",
+      "description": [
+        "This is the first of several challenges on sorting algorithms. Given an array of unsorted items, we want to be able to return a sorted array. We will see several different methods to do this and learn some tradeoffs between these different approaches. While most modern languages have built-in sorting methods for operations like this, it is still important to understand some of the common basic approaches and learn how they can be implemented.",
+        "Here we will see bubble sort. The bubble sort method starts at the beginning of an unsorted array and 'bubbles up' unsorted values towards the end, iterating through the array until it is completely sorted. It does this by comparing adjacent items and swapping them if they are out of order. The method continues looping through the array until no swaps occur at which point the array is sorted.",
+        "This method requires multiple iterations through the array and for average and worst cases has quadratic time complexity. While simple, it is usually impractical in most situations.",
+        "Instructions: Write a function bubbleSort which takes an array of integers as input and returns an array of these integers in sorted order. We've provided a helper method to generate a random array of integer values."
+      ],
+      "challengeSeed": [
+        "// generate a randomly filled array",
+        "let array = new Array();",
+        "(function createArray(size = 5) {",
+        "  array.push(+(Math.random() * 100).toFixed(0));",
+        "  return (size > 1) ? createArray(size - 1) : undefined;",
+        "})(12);",
+        "function bubbleSort(array) {",
+        "  // change code below this line",
+        "  // change code above this line",
+        "  return array;",
+        "};"
+      ],
+      "tail": [
+        "function isSorted(arr) {",
+        "  let check = (i) => (i == arr.length - 1) ? true : (arr[i] > arr[i + 1]) ? false : check(i + 1);",
+        "  return check(0);",
+        "};"
+      ],
+      "tests": [
+        "console.assert(typeof bubbleSort == 'function', 'message: bubbleSort is a function');",
```

Comment:
Oh no, that's actually left over from when I originally wrote this. I meant to have it updated to `assert` like the rest.